plugins/gatsby-transformer-garden-svg/tokens.json

```diff
@@ -0,0 +1,232 @@
+[
```

Comment:
Similar to my previous [comment](https://github.com/zendeskgarden/website/pull/352#pullrequestreview-890298772), I believe this information belongs co-located with and exported from `svg-icons`. This makes more sense from a maintenance and resusability POV. I would also like to take some time to go over the format together. It feels like these should be key-values, keyed by the icon prefix slug.

Reply:
However, before we take a run at the JSON placement/format, I'd like to get confirmation from @diana-oum on whether we're using Garden's sentence case for name [capitalization](https://garden.zendesk.com/content/grammar#capitalization).

Reply:
In Bethany's mock, it's sentence case! @exelarios

Reply:
Ah, thanks @diana-oum! I didn't catch the capitalization.

@jzempel 
Hmm, I will consider moving the tokens into `svg-icons` since it does make more sense to have them there. I'll have a look into how I would move the JSON into the `svg-icons`.

The JSON is formatted as an array of Objects instead of key-values is because the SVG itself can point to multiple tokens; thus the icon SVG (e.g `plus-stroke`) name isn't unique. I'm not entirely sure what you mean by `keyed by the icon prefix slug`?

Do you mean something like this? (e.g `info-stroke`)
```
"info": {
 "name": "Alert Info",
 "alt": ["information", "I", "circle"]
}
```

Reply:
yes, your example is the key-value format I'm suggesting.

I also think that once this gets over to `svg-icons` it might make more sense to export as a JS `const`. Once you get that work started, please connect directly with me so we can discuss. 🙏

Reply:
I tried that at first but realized the SVG name points to multiple token names. That's the reason why I went with an array of Objects. Not sure, if I'm overlooking anything?
<img width="789" alt="image" src="https://user-images.githubusercontent.com/30312964/158700032-8e24ee15-7640-4165-aad3-966a262ce5bc.png">

Reply:
I agree that the icons should be represented as an object and [normalized](https://redux.js.org/usage/structuring-reducers/normalizing-state-shape#designing-a-normalized-state). In this approach, an entity (icon) is keyed by its unique identifier (e.g. `info-stroke`). The full SVG icon file name should be used as the identifier since keys in a JavaScript object must be unique, and the object must account for both stroke and fill variants. Also, icon names (a new concept in Garden?) should be represented by an array since multiple names can belong to a unique icon.

I imagine the normalized object representation would be like:

```js
// In some central place, source of truth
const icons = {
  "chevron-left": {
    names: ["Hide Left", "Back"],
    tokens: []
  },
  "plus-stroke": {
    names: ["Add New"],
    tokens: []
  },
  "info-stroke": {
    names: ["Alert Info"],
    tokens: ["information", "i", "circle"]
  },
  ...restIcons
};
```

Given the normalized representation, UI data can then be derived from it:

```js
import icons from 'source-of-truth';

const iconDataForGatsby = Object.keys(icons).reduce((accumulator, identifier) => {
  const { names, tokens } = icons[identifier];
  return [
    ...accumulator,
    ...names.map((name) => ({
      icon: identifier,
      tokens,
      name
    }))
  ];
}, []);
```

Reply:
Thanks @hzhu for thinking through this and sharing your approach. I love how you used the `reduce` function. 😆

Just synced up with @jzempel and he bought up a great point that there are currently three sources of truth; Figma, Google Sheets, and JSON if we end up with our current approach. So the new approach is directly reading tokens off from Figma where that can be the source-of-truth for the icon tokens.

Since there's a many-to-one relationship with the icon/slug, we're going with the naming of the tokens to the key identifier for the token.

JZ proposed the data be structured something like this:
```js
"add new": {
 "icon": "plus-stroke",
 "synonyms": []
}
```

Reply:
I see, so the base schema is a token, not an icon. Essential each key (e.g. add new) is mapped to a token identifier. Yeah that structure makes sense to me.